### PR TITLE
Log Parser - Version 1.0 - Preliminary

### PR DESCRIPTION
**Pull request by udi-speedb**

### CODE REVIEW

**Comment by ofriedma** [comment]
histograms.csv

**Comment by udi-speedb** [comment]
@ofriedma Thanks for the review and your comments.
I will update the tool and address all of your comments.

**Review by ofriedma — CHANGES_REQUESTED** [review]
1. Long output imo should contain only json output.
There are more outputs in the end which prevent easy use of jq after the output.
2. rmdir to the directory is not ok line 104 log_parser.py - too risky, we should maybe overwrite existing files or creating them with -<number>, if the dir exists we should just output there
3. On first run it looks like there is an error when the output dir is not output_files:
FileNotFoundError: [Errno 2] No such file or directory: 'output_files/abc.json'
The command was:
python3 log_parser.py -c short -j abc.json -o ./mylogs ../Downloads/logudi
4. Better name convention for histograms_1 and _2, maybe human_readable, such as:
histograms.csv
histograms_human_readable.csv

**Comment by udi-speedb** [comment]
@ofriedma - I have pushed an update that addresses all of your comment. Please review.

File: README.md [hunk 1]
```diff
@@ -1 +1,142 @@
...
+
+
+
+
+<!-- GETTING STARTED -->
+## Getting Started
+The tool runs on a single log file and generates one or more outputs:
+1. A short summary printed to the console
+2. A detailed summary in a JSON format
+3. The detailed JSON file printer to the console. This may be filtered
```
Comment (Yuval-Ariel) [inline README.md:20]: "a detailed JSON file printed to the console" ?

File: README.md [hunk 2]
```diff
@@ -1 +1,142 @@
...
+summary.
+
+The outputs are placed in a separate folder (the folder is DELETED if 
+already exists). By default, "output_files" will be the name of that folder 
+but the user may override that.
+
+Running the tool without any parameters will allow users to view the 
+possible flags the tool supports:
+   ```sh
+   pytho3 log_parser.py
```
Comment (Yuval-Ariel) [inline README.md:40]: "python3" and below as well

File: README.md [hunk 3]
```diff
@@ -1 +1,142 @@
...
+   pytest
+   ```
+   
+
+<!-- USAGE EXAMPLES -->
+## Usage
+
+The repo contains a sample file that may be used to run the tool:
+   ```sh
+   pytho3 test/input_files/LOG_speedb
```
Comment (Yuval-Ariel) [inline README.md:91]: python3 here as well

File: README.md (hunk 1 above)
Comment (udi-speedb) [reply README.md:20]: Yes, @ofriedma proposed doing that. The detailed information in the json contains tons of information. Reading it on the console is not so practical or convenient. However, there are widely-available command line tools that are able to parse json (e.g., JQ) and allow proficient console users to easily filter the json on the console (only those would print the detailed information to the console anyway)

File: README.md [hunk 4]
```diff
@@ -1 +1,142 @@
-# log-parser
+## Speedb-Log-Parser: A tool for parsing RocksDB and Speedb Log Files
+
+<!-- ABOUT THE PROJECT -->
+## About The Project
+Speedb's Log Parser is a tool that may be used to parse and process 
+Speedb and RocksDB log files.
+
+The tool extracts useful information from these logs and aids users in 
+gaining insights about their systems with ease. It is expected to be valuable tool
```
[the remark this answers is not in the document]
Comment (udi-speedb) [reply README.md:9]: Isn't it?

File: README.md (hunk 2 above)
Comment (udi-speedb) [reply README.md:40]: Fixed

**Comment by ofriedma** [comment]
@udi-speedb
running:` python3 log_parser.py -c short -j abc.json -o /home/rook/log-parser ../Downloads/logudi`
cause: 
```
Traceback (most recent call last):
  File "/home/or/log-parser/log_parser.py", line 270, in <module>
    main()
  File "/home/or/log-parser/log_parser.py", line 227, in main
    output_folder = prepare_output_folder(output_folder)
  File "/home/or/log-parser/log_parser.py", line 115, in prepare_output_folder
    updated_output_path.mkdir()
  File "/usr/lib/python3.10/pathlib.py", line 1175, in mkdir
    self._accessor.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: 'log-parser/log_parser_0001'
```

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Yuval-Ariel) [reply README.md:9]: just a spelling fix ;) add the "a"

File: README.md (hunk 1 above)
Comment (Yuval-Ariel) [reply README.md:20]: also, just a spelling fix

**Comment by udi-speedb** [comment]
I have pushed a commit with some updates following your comments and some minor additional issues:
- Fixed typos in README.md
- No console messages about generated files to allow tools like JQ to run properly and have a consistent output
- I am now generating a sub-folder under the output folder in the format log_parser_XXXX where XXXX are 4 digits. The number increases by 1 every run and wraps around, so, if the user specifies -o output, the files will be under output_folder/log_parser_0001 in the first run, output_folder/log_parser_0002 in the seconds run, and so on and so forth. If you feel this is not a good approach, please propose another.
- Miscellaneous minor cleanups

**Comment by udi-speedb** [comment]
And another comment with some additional updates / fixes to the README.md file

**Review by ofriedma — APPROVED** [review]
LGTM

**Comment by Yuval-Ariel** [comment]
Compaction Style         : kCompactionStyleLevel
Compression              : NoCompression
Filter-Policy            : bloomfilter

is there a reason to print the above three even when these are printed per column family?

**Comment by Yuval-Ariel** [comment]
DB Size                  : 32.6 MB
Num Keys                 : 48,033,559

from what i can tell, Num Keys is taken from cumulative writes and it seems to me a bit misleading. 
a user can self deduce the amount of keys actually in the db by its size and avg size.
so if we intend to provide useful information about how much has been ingested, then some adjustments are required IMO.

**Comment by udi-speedb** [comment]
> Compaction Style : kCompactionStyleLevel Compression : NoCompression Filter-Policy : bloomfilter
> 
> is there a reason to print the above three even when these are printed per column family?

I agree that as it is, you are correct and we should remove these fields as you suggest.

**Comment by udi-speedb** [comment]
> Cumulative writes

I am showing 0 keys since I am taking the number from the statistics. However, in this case there are no statistics.
I should probably switch to using the "Cumulative Writes" as this is almost certain to appear in the logs

**Comment by udi-speedb** [comment]
I have updated the branch following your comments. Please review. Thanks

**Pull request closed without merging**